specs/Standards/Contracts/SelfIdentification.md

```diff
@@ -0,0 +1,207 @@
+   webAppUrl?: string; //web URL where the DApp can be accessed e.g. http://Narwallets.com/div-pool/
+   auditorAccountId?: string=""; //near account of the auditor of the code
+}
+```
+
+### Details on the fields:
+- **dataVersion**. In future iterations of this NEP, new fields migth be added to `ContractInfo`, this number will be increased. All `dataVersions` are backwards compatible, meaning `dataVersion=2` **can only add fields** to `dataVersion=1`'s structure
+
+### Use Cases 
+
+Alice wants to trade only with audited NEP-21 tokens. She receives an account name as a token source. Before interacting with the account her code can:
+
+* perform basic account validation (check that there are no full-access keys on the contract)
+
+* call `get_contract_info`
+
+* verify the contract supports NEP-21 `standards.include("NEP-21")`
+
+* use `auditorAccountId` to check the contracts code hash and audit info (depending on the auditor's mechanism to do that)
+
+If all the above is correct, her code adds the contract as an audited NEP-21 Token and the user can interact with it via `contractInfo.webAppUrl`
+
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Reference Implementations:
+
+#### Rust:
+```rust
+// ---------------------------------
+// -- CONTRACT Self Identification --
+// ---------------------------------
+// [NEP-129](https://github.com/nearprotocol/NEPs/pull/129)
+// see also pub fn get_contract_info
+pub const CONTRACT_NAME: &str = "diversifying staking pool";
+pub const CONTRACT_VERSION: &str = "0.1.0";
+pub const DEVELOPERS_ACCOUNT_ID: &str = "developers.near"; 
+pub const DEFAULT_WEB_APP_URL: &str = "http://div-pool.narwallets.com";
+pub const DEFAULT_AUDITOR_ACCOUNT_ID: &str = "auditors.near";
+
+/// NEP-129 get information about this contract
+/// returns JSON string according to [NEP-129](https://github.com/nearprotocol/NEPs/pull/129)
+/// Rewards fee fraction structure for the staking pool contract.
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+#[allow(non_snake_case)]
+pub struct NEP129Response {
+    pub dataVersion:u16,
+    pub name:String,
+    pub version:String,
+    pub developersAccountId:String,
+    pub source:String,
+    pub standards:Vec<String>,
+    pub webAppUrl:Option<String>,
+    pub auditorAccountId:Option<String>,
+}
+
+/// NEP-129 get information about this contract
+/// returns JSON string according to [NEP-129](https://github.com/nearprotocol/NEPs/pull/129)
+pub fn get_contract_info(&self) -> NEP129Response {
+    return NEP129Response {
+        dataVersion:1,
+        name: CONTRACT_NAME.into(),
+        version:CONTRACT_VERSION.into(),
+        developersAccountId:DEVELOPERS_ACCOUNT_ID.into(),
+        source:"https://github.com/Narwallets/diversifying-staking-pool".into(), 
+        standards:vec!("NEP-129".into()),  
+        webAppUrl:self.web_app_url.clone(),
+        auditorAccountId:self.auditor_account_id.clone()
+    }
+}
```

Comment:
I found the current code distracting. I think either pseudo code is better (potentially moving this implementation to assets) and making sure it compiles (if copy pasted properly in the project).

Otherwise, let's make sure the code is properly formatted, get rid of global variables given they are used only once, properly update comments so they are more useful.